### PR TITLE
rovr_control refactor

### PR DESCRIPTION
**Pull request by Isopod00**
This PR is finished but I should probably test it on the robot before merging.

### CODE REVIEW

File: src/rovr_control/rovr_control/autonomous_procedures.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+class autonomous_procedure:
+    """ This class is a template used to define an autonomous procedure. """
+    def __init__(self, rovr_control_instance: Node, name: str, procedure_logic: Callable) -> None:
+        self.name = name
+        self.node_instance = rovr_control_instance
+        self.procedure_logic = procedure_logic
+    async def run(self):
+        self.node_instance.stop_all_subsystems() # Stop all subsystems
+        self.node_instance.state = self.node_instance.states[self.name]
+        print(f"\nStarting {self.name} Procedure!")
```
Comment (jmblixt3) [inline src/rovr_control/rovr_control/autonomous_procedures.py:18]: You are using print statements, shouldn't ros logger be used instead

Comment (jmblixt3) [inline src/rovr_control/rovr_control/autonomous_procedures.py:21]: You are using print statements, shouldn't ros logger be used instead

**Review by jmblixt3 — COMMENTED** [review]
Please change these if possible

File: src/rovr_control/rovr_control/autonomous_procedures.py (hunk 1 above)
Comment (Isopod00) [reply src/rovr_control/rovr_control/autonomous_procedures.py:21]: In the dev container the ROS logger didn't work here, but I will test on the robot to see if it works there

**Pull request closed without merging**